content/advent-2016/saga-go-dependency-management.md

```diff
@@ -0,0 +1,106 @@
++++
+date = "2016-12-13T00:00:00"
+author = [ "Sam Boyer (@sdboyer)" ]
+title = "The Saga of Go Dependency Management"
+series = ["Advent 2016"]
++++
+
+# The Saga of Go Dependency Management
+
+The Go community is on the cusp of some major shifts in the way we handle dependencies. These shifts are a long time coming, and involve the work and effort of dozens, if not hundreds, of people. Six to twelve months from now, the Go dependency management landscape is likely to look very different.
+
+For those who haven't been following closely - or, honestly, even for those who have - it can be difficult to keep track of what's going on, and where things are going. So, as part of the Gopher advent series, we thought we'd tie up Go package management past, present, and future in a nice big bow.
+
+## The road so far
+
+Anyone who takes a stroll down the memory lane of Go package management will find ample clumps of ripped-out hair strewn about. The process has been long and frustrating, and there's far too much to cover in detail here. But the history is important to where we are now, so a high-level timeline is worth it.
+
+`go get` was, and is, the only official tooling for retrieving Go code and placing it on disk. It has served Go users and developers alike since the release of Go 1.0. Hand-in-hand with `go get` has been the general recommendation from the Go Team on how to keep everything working well: "write backwards-compatible code." That's a slightly-rephrased version of the Go 1 guarantee. This pairs well with `go get`, as `go get`'s behavior could only be correct if the following were true:
+
+1. All code is always backwards compatible
+2. There are no fixed points/releases
+
+_(to be clear, the authors of `go get` do not necessarily believe this is good or right - just that when these things aren't true, it's not `go get`'s problem.)_
+
+In an ecosystem structured around `go get`, the most obvious problem was reproducibility: because `go get` fetches the latest code for any packages not already on disk, it's impossible to guarantee that your users  - or your teammates, or build system - will have the same versions of your software's dependencies that you do. Being the most gaping wound, reproducibility was the main problem that tooling has sought to address, especially early on. [godep](https://github.com/tools/godep) - still the most widely used tool today - emerged just thirteen months after the 1.0 release of Go, with a laser focus on reproducible builds.
```

Comment:
Maybe instead of:

> Being the most gaping wound, reproducibility was the main problem that tooling has sought to address, especially early on.

Try:

> Most third-party Go dependency tools focused on letting users create reproducible builds.